hw5/task5_3/task5_3/Program.fs

```
@@ -0,0 +1,110 @@
+            let newRecord = { Name = newData.[0]; 
+                              Phone = newData.[1]}
+            loadPhoneBook (newRecord::acc)
+        else
+            reader.Close()
+            acc
+    printfn "Data loaded"
+    loadPhoneBook []
+
+let rec phoneBookInterface (records : List<record>) =
+    printfn "Enter the command "
+    let input = Console.ReadLine()
+    let parsed, result = System.Int32.TryParse(input)
+    if (not parsed || result < 1 || result > 7) then
+        printfn "Incorrect input, try again"
+        phoneBookInterface records
+    else
+        match result with
+        | 1 -> printfn "Shut down"
+        | 2 -> printfn "Enter the name: "
+               let newName = Console.ReadLine()
+               printfn "Enter the phone number: "
+               let newPhone = Console.ReadLine()
+               phoneBookInterface <| addRecord records newName newPhone
+        | 3 -> printfn "Enter the name to search: "
+               let name = Console.ReadLine()
+               findByName records name
+               phoneBookInterface records
+        | 4 -> printfn "Enter the phone number to search: "
+               let phone = Console.ReadLine()
+               findByPhone records phone
+               phoneBookInterface records
+        | 5 -> printDB records
+               phoneBookInterface records
+        | 6 -> printfn "Enter the target file path: "
+               let path = Console.ReadLine()
+               saveToFile records path
+               phoneBookInterface records
+        | 7 -> printfn "Enter the target file path: "
+               let path = Console.ReadLine()
+               try
+                   let loaded = loadFromFile path
+                   phoneBookInterface loaded
+               with
+               | :? FileNotFoundException ->
+                   printfn "File not found"
+                   phoneBookInterface records
+               | _ -> 

```

Лучше не ловить что попало. Неизвестная ошибка --- ок, дайте программе упасть. Fail fast и всё такое